bindings/pyroot_experimental/PyROOT/src/DeclareCallablePyz.cxx

```diff
@@ -0,0 +1,663 @@
+// Author: Stefan Wunsch, Enric Tejedor CERN  04/2019
+// Original PyROOT code by Wim Lavrijsen, LBL
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "Python.h"
+
+#include "CPyCppyy.h"
+#include "PyROOTPythonize.h"
+#include "CPPInstance.h"
+#include "Utility.h"
+#include "TInterpreter.h"
+#include "TInterpreterValue.h"
+
+#include <sstream>
+
+
+// Init of class used as decorator to create generic C++ wrapper
+// The init parses the arguments passed to the decorator.
+PyObject* GenericCallableImpl_init(PyObject * /*self*/, PyObject *args)
+{
+   // Parse arguments and extract input and output types and an optional verbose flag
+   if (!PyTuple_Check(args)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Invalid tuple.");
+      return NULL;
+   }
+
+   const auto size = PyTuple_Size(args);
+   if (!(size == 3 || size == 4)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly two arguments (list of input types, return type, optional name).");
```

Comment:
Says "expect exactly two arguments" but then specifies three, we could say "expects up to three arguments:x, y, z (optional)"

Reply:
Ah, my bad, I'll fix this. This is simply wrong form my side.